ydb/core/mind/node_broker__decommission_node.cpp

```diff
@@ -0,0 +1,85 @@
+#include "node_broker_impl.h"
+#include "node_broker__scheme.h"
+
+#include <ydb/core/base/appdata.h>
+#include <ydb/core/protos/counters_node_broker.pb.h>
+
+namespace NKikimr {
+namespace NNodeBroker {
+
+using namespace NKikimrNodeBroker;
+
+class TNodeBroker::TTxDecommissionNode : public TTransactionBase<TNodeBroker> {
+public:
+    TTxDecommissionNode(TNodeBroker *self, TEvNodeBroker::TEvDecommissionRequest::TPtr &ev)
+        : TBase(self)
+        , Event(ev)
+    {
+    }
+
+    TTxType GetTxType() const override { return TXTYPE_DECOMMISSION_NODE; }
+
+    bool Error(TStatus::ECode code,
+               const TString &reason,
+               const TActorContext &ctx)
+    {
+        const auto &rec = Event->Get()->Record;
+        auto host = rec.GetHost();
+        auto port = rec.GetPort();
+        LOG_ERROR_S(ctx, NKikimrServices::NODE_BROKER,
+                    "Cannot Decommission node " << host << ":" << port << ": " << code << ": " << reason);
+
+        Response->Record.MutableStatus()->SetCode(code);
+        Response->Record.MutableStatus()->SetReason(reason);
+
+        return true;
+    }
+
+    bool Execute(TTransactionContext &txc, const TActorContext &ctx) override
+    {
+        Y_UNUSED(txc);
+
+        auto &rec = Event->Get()->Record;
+        auto host = rec.GetHost();
+        ui16 port = (ui16)rec.GetPort();
+        TString addr = rec.GetAddress();
```

Comment:
Тут лишняя копия, можно без нее:
```suggestion
        const auto& addr = rec.GetAddress();
```